src_Core/CPU/CPU.bsv

```diff
@@ -866,7 +984,21 @@ module mkCPU (CPU_IFC);
 		     rg_next_seq <= stage1.out.data_to_stage2.instr_seq + 1;
 `endif
 		     redirect = True;
+`ifdef PERFORMANCE_MONITORING
+		     events.evt_REDIRECT = True;
+`endif
 		  end
+`ifdef PERFORMANCE_MONITORING
+		  if (   (stage1.out.data_to_stage2.op_stage2 == OP_Stage2_ST)
+		      && (stage1.out.data_to_stage2.mem_width_code == w_SIZE_CAP)   ) begin
```

Comment:
What about AMOs (both AMOSWAP.C and SC.C)? How is this useful to count if we don't know about how many _non-capability_ stores there are as it's just a number with no context? I'm not even sure how useful to software it is then, because memcpy and memset will use capability loads/stores whenever aligned enough so you'll see a lot of these even in hybrid code.